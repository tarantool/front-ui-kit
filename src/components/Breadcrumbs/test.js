import React from 'react';
import renderer from 'react-test-renderer';

import { Breadcrumbs } from '.';

const noop = () => void 0;

describe('Breadcrumbs', () => {
  const breadcrumbs = [
    {
      title: '@tarantool.io/ui-kit',
      path: '/tarantool',
    },
    {
      title: 'UI Components',
      path: '/section-ui-components',
    },
    {
      title: 'Breadcrumb',
      path: '/breadcrumb',
    },
    {
      title: 'OverflowList',
      path: '/overflow-list',
    },
    {
      title: 'ResizeSensor',
      path: '/resize-sensor',
    },
  ];

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <>
          <Breadcrumbs items={breadcrumbs} appName={'Project name'} onLinkClick={noop} />
        </>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly long name', () => {
    const tree = renderer
      .create(
        <>
          <Breadcrumbs
            items={breadcrumbs}
            appName={'Project name-2948474770249-999-20299547789123123141'}
            onLinkClick={noop}
          />
        </>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('block with small width', () => {
    const tree = renderer
      .create(
        <>
          <div style={{ width: '300px' }}>
            <Breadcrumbs
              items={breadcrumbs}
              appName={'Project name-2948474770249-999-20299547789123123141'}
              onLinkClick={noop}
            />
          </div>
        </>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
