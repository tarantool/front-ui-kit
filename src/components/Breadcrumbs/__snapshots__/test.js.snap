// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs block with small width 1`] = `
<div
  style={
    Object {
      "width": "300px",
    }
  }
>
  <div
    className="css-8vahvp"
  >
    <span
      className="css-1u31squ"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    >
      Project name-2948474770249-999-202995477...
    </span>
    <div
      className="css-vzwfet"
    >
      <span
        className="css-sw5zrb"
      >
        /
      </span>
      <a
        className="css-1994ztc"
        href="/tarantool"
        onClick={[Function]}
      >
        @tarantool.io/ui-kit
      </a>
      <span
        className="css-sw5zrb"
      >
        /
      </span>
      <a
        className="css-1994ztc"
        href="/section-ui-components"
        onClick={[Function]}
      >
        UI Components
      </a>
      <span
        className="css-sw5zrb"
      >
        /
      </span>
      <a
        className="css-1994ztc"
        href="/breadcrumb"
        onClick={[Function]}
      >
        Breadcrumb
      </a>
      <span
        className="css-sw5zrb"
      >
        /
      </span>
      <a
        className="css-1994ztc"
        href="/overflow-list"
        onClick={[Function]}
      >
        OverflowList
      </a>
      <span
        className="css-sw5zrb"
      >
        /
      </span>
      <a
        className="css-1994ztc"
        href="/resize-sensor"
        onClick={[Function]}
      >
        ResizeSensor
      </a>
      <div
        className="css-11rp940"
      />
    </div>
  </div>
</div>
`;

exports[`Breadcrumbs renders correctly 1`] = `
<div
  className="css-8vahvp"
>
  <span
    className="css-4haz1u"
  >
    Project name
  </span>
  <div
    className="css-vzwfet"
  >
    <span
      className="css-sw5zrb"
    >
      /
    </span>
    <a
      className="css-1994ztc"
      href="/tarantool"
      onClick={[Function]}
    >
      @tarantool.io/ui-kit
    </a>
    <span
      className="css-sw5zrb"
    >
      /
    </span>
    <a
      className="css-1994ztc"
      href="/section-ui-components"
      onClick={[Function]}
    >
      UI Components
    </a>
    <span
      className="css-sw5zrb"
    >
      /
    </span>
    <a
      className="css-1994ztc"
      href="/breadcrumb"
      onClick={[Function]}
    >
      Breadcrumb
    </a>
    <span
      className="css-sw5zrb"
    >
      /
    </span>
    <a
      className="css-1994ztc"
      href="/overflow-list"
      onClick={[Function]}
    >
      OverflowList
    </a>
    <span
      className="css-sw5zrb"
    >
      /
    </span>
    <a
      className="css-1994ztc"
      href="/resize-sensor"
      onClick={[Function]}
    >
      ResizeSensor
    </a>
    <div
      className="css-11rp940"
    />
  </div>
</div>
`;

exports[`Breadcrumbs renders correctly long name 1`] = `
<div
  className="css-8vahvp"
>
  <span
    className="css-1u31squ"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    Project name-2948474770249-999-202995477...
  </span>
  <div
    className="css-vzwfet"
  >
    <span
      className="css-sw5zrb"
    >
      /
    </span>
    <a
      className="css-1994ztc"
      href="/tarantool"
      onClick={[Function]}
    >
      @tarantool.io/ui-kit
    </a>
    <span
      className="css-sw5zrb"
    >
      /
    </span>
    <a
      className="css-1994ztc"
      href="/section-ui-components"
      onClick={[Function]}
    >
      UI Components
    </a>
    <span
      className="css-sw5zrb"
    >
      /
    </span>
    <a
      className="css-1994ztc"
      href="/breadcrumb"
      onClick={[Function]}
    >
      Breadcrumb
    </a>
    <span
      className="css-sw5zrb"
    >
      /
    </span>
    <a
      className="css-1994ztc"
      href="/overflow-list"
      onClick={[Function]}
    >
      OverflowList
    </a>
    <span
      className="css-sw5zrb"
    >
      /
    </span>
    <a
      className="css-1994ztc"
      href="/resize-sensor"
      onClick={[Function]}
    >
      ResizeSensor
    </a>
    <div
      className="css-11rp940"
    />
  </div>
</div>
`;
